storm-core/test/jvm/org/apache/storm/LocalStateTest.java

```diff
@@ -0,0 +1,54 @@
+package org.apache.storm;
+
+import org.apache.storm.generated.GlobalStreamId;
+import org.apache.storm.testing.TmpPath;
+import org.apache.storm.utils.LocalState;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.IOException;
+
+public class LocalStateTest {
+
+    @Test
+    public void testLocalState() throws IOException{
+        TmpPath dir1_tmp = new TmpPath();
```

Comment:
Let's use try-with-resource to make it compatible with `with-open` in Clojure. It should cover dir1_tmp and dir2_tmp.